### PR TITLE
Fix tensorflow fft2 backend

### PR DESCRIPTION
**Pull request by AbdullahSabry**
Fixed the backend for `fft2` in tensorflow backend related to the PRs #22366 and #22309 where the tests were failing.

### CODE REVIEW

**Comment by AbdullahSabry** [comment]
Hey @abdulasiraj 

I believe the issue with the function is fixed with the added code. It works with both the PRs mentioned above. Check out the code whenever. Thanks!

File: ivy/functional/backends/tensorflow/experimental/layers.py [hunk 1]
```diff
@@ -992,35 +1043,20 @@ def fft2(
...
-        x_complex = tf.cast(x_new, tf.complex128)
-        tf_fft2 = tf.transpose(tf.signal.fft2d(x_complex, name=operation_name))
+        result = tf.signal.fft2d(x_complex)
+
+    result = _fft2_norm(result, s, dim, norm)
 
-    # Apply the same normalization as 'backward' in NumPy
-    tf_fft2 = _fft2_norm(tf_fft2, s, dim, norm)
-    return tf_fft2
+    if out is not None:
```
Comment (abdulasiraj) [inline ivy/functional/backends/tensorflow/experimental/layers.py:1055]: Hi @AbdullahSabry,
`out` argument will be handled by `handle_out_argument` decorator. so, you no need to handle this. Please remove this block as it's reduntant.
Thanks

**Review by abdulasiraj — APPROVED** [review]
Everything else looks good to me. just remove the out block and let me know, I'll merge it. thankls!

**Review by abdulasiraj — CHANGES_REQUESTED** [review]
Hi @AbdullahSabry 
tests are failing locally, can you please double check what's the problem is?
```
Failed: [undefined]AssertionError: returned dtype = complex64, ground-truth returned dtype = complex128
```
P.S: problem is related to tensorflow backend

**Review by abdulasiraj — APPROVED** [review]

**Pull request merged**